### DISCUSSION TITLE
Best practice for webhook management

### DISCUSSION THREAD

**Discussion by matthewhaworth** [Kyon147/laravel-shopify#235] [Q&A]
Hello! I'm new to this world so excuse my naivety.

I can see that by default you set your webhooks up with Shopify when a store installs your app; that is how this module works here. However, how might I handle adding webhooks to shops _after_ they have installed the app later down the line? Say for example initially I only require a webhook when an order is created, and I release my app. Later down the line I add a feature that requires me to observe refunds also. What's the best practice approach to doing that?

**Comment by booni3** [comment]
```php
class SyncWebhooksAction
{
    public function execute(User $shop): void
    {
        $shopId = ShopId::fromNative($shop->id);
        $webhooks = config('shopify-app.webhooks');
        WebhookInstaller::dispatch($shopId, $webhooks);
    }
}
```

**Reply by Kyon147** [reply]
You can just add new webhooks into the config file, then the next time a store is passed through the /authenticate any new webhooks will be installed.

Otherwise, you could programmatically set them up as @booni3 suggests or just push customers through /authenticate yourself as well.
